data_override/data_override.F90

```diff
@@ -569,120 +134,93 @@ subroutine data_override_unset_domains(unset_Atm, unset_Ocean, &
-        UGdomain = lnd_domainUG
-     case default
-        call mpp_error(FATAL,'error in data_override get_domain')
+  select type(data)
+    type is (real(r4_kind))
+      call data_override_0d_r4(gridname,fieldname_code,data,time,override,data_index)
+    type is (real(r8_kind))
+      call data_override_0d_r8(gridname,fieldname_code,data,time,override,data_index)
+    class default
+      call mpp_error(FATAL, "data_override_0d: Unsupported data type")
   end select
-!  if(UGdomain .EQ. NULL_DOMAINUG) call mpp_error(FATAL,'data_override: failure in get_domain')
-  if(present(comp_domain)) &
-     call mpp_get_UG_SG_domain(UGdomain,SGdomain)
-     call mpp_get_compute_domain(SGdomain,comp_domain(1),comp_domain(2),comp_domain(3),comp_domain(4))
-end subroutine get_domainUG
-!===============================================================================================
+end subroutine data_override_0d
 
 !> @brief This routine performs data override for 2D fields; for usage, see data_override_3d.
 subroutine data_override_2d(gridname,fieldname,data_2D,time,override, is_in, ie_in, js_in, je_in)
   character(len=3), intent(in) :: gridname !< model grid ID
   character(len=*), intent(in) :: fieldname !< field to override
   logical, intent(out), optional :: override !< true if the field has been overriden succesfully
   type(time_type), intent(in) :: time !<  model time
-  real, dimension(:,:), intent(inout) :: data_2D !< data returned by this call
+  class(*), dimension(:,:), intent(inout) :: data_2D !< data returned by this call
   integer,           optional,  intent(in) :: is_in, ie_in, js_in, je_in
-!  real, dimension(size(data_2D,1),size(data_2D,2),1) :: data_3D
-  real, dimension(:,:,:), allocatable ::  data_3D
-  integer       :: index1
-  integer       :: i
 
-!1  Look  for the data file in data_table
-  if(PRESENT(override)) override = .false.
-  index1 = -1
-  do i = 1, table_size
-     if( trim(gridname) /= trim(data_table(i)%gridname)) cycle
-     if( trim(fieldname) /= trim(data_table(i)%fieldname_code)) cycle
-     index1 = i                               ! field found
-     exit
-  enddo
-  if(index1 .eq. -1) return  ! NO override was performed
-
-  allocate(data_3D(size(data_2D,1),size(data_2D,2),1))
-  data_3D(:,:,1) = data_2D
-  call data_override_3d(gridname,fieldname,data_3D,time,override,data_index=index1,&
-                       is_in=is_in,ie_in=ie_in,js_in=js_in,je_in=je_in)
-
-  data_2D(:,:) = data_3D(:,:,1)
-  deallocate(data_3D)
+  select type(data_2D)
+    type is (real(r4_kind))
+      call data_override_2d_r4(gridname,fieldname,data_2D,time,override, is_in, ie_in, js_in, je_in)
+    type is (real(r8_kind))
+      call data_override_2d_r8(gridname,fieldname,data_2D,time,override, is_in, ie_in, js_in, je_in)
+    class default
+      call mpp_error(FATAL, "data_override_2d: Unsupported data type")
+  end select
 end subroutine data_override_2d
 
 !> @brief This routine performs data override for 3D fields
```

Comment:
The documentation needs to be updated

Reply:
This subroutine was removed in b89c8db621164eef22e0d9220a2046a4a1ceaa98.